src/libraries/System.Reflection.Metadata/src/System/Reflection/Metadata/BlobContentId.cs

```diff
@@ -66,16 +66,12 @@ public static unsafe BlobContentId FromHash(byte[] hashCode)
                 throw new ArgumentException(SR.Format(SR.HashTooShort, minHashSize), nameof(hashCode));
             }
 
-            Guid guid = default(Guid);
-            byte* guidPtr = (byte*)&guid;
-            for (var i = 0; i < BlobUtilities.SizeOfGuid; i++)
-            {
-                guidPtr[i] = hashCode[i];
-            }
-
             // modify the guid data so it decodes to the form of a "random" guid ala rfc4122
-            guidPtr[7] = (byte)((guidPtr[7] & 0x0f) | (4 << 4));
-            guidPtr[8] = (byte)((guidPtr[8] & 0x3f) | (2 << 6));
+            byte[] guidBytes = new byte[16];
```

Comment:
Anywhere we're adding allocations that we weren't previously allocating is a red flag.  We shouldn't be doing this here.

Reply:
Changed to keep the logic on little-endian and just add extra swaps for the big-endian code path.